scripts/metricsgen/metricsgen.go

```diff
@@ -272,6 +266,28 @@ func extractTypeName(e ast.Expr) string {
 	return strings.TrimPrefix(path.Ext(types.ExprString(e)), ".")
 }
 
+func extractHelpMessage(cg *ast.CommentGroup) string {
+	if cg == nil {
+		return ""
+	}
+	var help string
+	for i, c := range cg.List {
+		str := strings.TrimPrefix(c.Text, "//")
+		if len(str) == 0 {
+			continue
+		}
+		mt := strings.TrimPrefix(str, "metrics:")
+		if len(mt) < len(str) {
+			return mt
+		}
+		help += strings.TrimPrefix(str, " ")
+		if i < len(cg.List)-1 {
+			help += " "
+		}
+	}
+	return help
```

Comment:
Fine as written. Some possible simplifications:

```suggestion
	var help []string
	for _, c := range cg.List {
		mt := strings.TrimPrefix(str, "//metrics:")
		if mt != c {
			return strings.TrimSpace(mt)
		}
		help = append(help, strings.TrimSpace(strings.TrimPrefix(c, "//")))
	}
	return strings.Join(help, " ")
```